mitodl/mitxonline
# Restrict single active product per course ID

#### Pre-Flight checklist


#### What are the relevant tickets?
(Required)

#### What's this PR do?
There was an error occurring when a user was applying for Flexible Pricing on a Course which was associated with two Product that were `active`.  https://sentry.io/organizations/mit-office-of-digital-learning/issues/3426908655/?environment=rc&project=5864687&query=is%3Aunresolved+&statsPeriod=14d

This PR adds a check when retrieving the Product related to the Course to ensure that it has `is_active` == `true`.

This PR also adds a migration to the Product model to ensure that only a single Product can be active if multiple Products are linked to the same Course ID.

#### How should this be manually tested?

- Login to the Django admin
- Create a Course
- Create a Product record linked to the Course that is active.
- Create a second Product record linked to the same course that is also active.
- You should receive an error message.  Right now this is a 500 error.  I would like to make it nicer for the user, but I'm not sure of how to do that.
- Create a second Product record linked to the same course that is not active.  This should succeed.
![Screen Shot 2022-08-03 at 9 52 53 AM](https://user-images.githubusercontent.com/8311573/182625510-f83eeb84-42ea-4999-897c-7a1ce817ea6c.png)
